Mathlib/RingTheory/Artinian.lean

```diff
@@ -498,4 +498,26 @@ instance isMaximal_of_isPrime (p : Ideal R) [p.IsPrime] : p.IsMaximal :=
       localization_surjective (nonZeroDivisors (R ⧸ p)) (FractionRing (R ⧸ p))⟩).isField _
     <| Field.toIsField <| FractionRing (R ⧸ p)
 
+variable (R) in
+lemma prime_spectrum_finite : {I : Ideal R | I.IsPrime}.Finite := by
+  let Spec := {I : Ideal R | I.IsPrime}
+  obtain ⟨_, ⟨s, rfl⟩, H⟩ := set_has_minimal
+    (range (Finset.inf · Subtype.val : Finset Spec → Ideal R)) ⟨⊤, ∅, by simp⟩
+  refine ⟨⟨s, fun p ↦ ?_⟩⟩
+  obtain ⟨q, hq1, hq2⟩ : ∃ q ∈ s, q ≤ p
+  · classical
+    simpa only [mem_setOf_eq, coe_setOf, ge_iff_le, Finset.le_inf_iff, Subtype.coe_le_coe,
+      Subtype.forall, inf_lt_right, not_not, Ideal.IsPrime.inf_le' (hp := p.2)]
+      using H (p ⊓ s.inf Subtype.val) ⟨insert p s, by simp⟩
+  rwa [← Subtype.ext_iff.mpr <| @isMaximal_of_isPrime (p := q.1) _ _ q.2 |>.eq_of_le p.2.1 hq2]
+
+variable (R) in
+/--
+[Stacks Lemma 00J7](https://stacks.math.columbia.edu/tag/00J7)
+-/
+lemma maximal_ideals_finite : {I : Ideal R | I.IsMaximal}.Finite := by
+  convert prime_spectrum_finite R using 3
+  exact ⟨Ideal.IsMaximal.isPrime, fun h => isMaximal_of_isPrime _⟩
```

Comment:
maybe add a isPrime_iff_isMaximal lemma for artinian rings.

Reply:
done